app/__init__.py

```diff
@@ -53,19 +103,82 @@ def home():
 
 @app.route("/flixlist/<menu_item>")
 def flixlist(menu_item):
-    rows = 10
-    return render_template("flixlist.html", title="FlixList", menu_item=menu_item, rows=rows)
+
+    # Check search button
+    query = request.args.get("search")
+    if query == None:
+        # Get Movies/TV Shows info
+        if menu_item == "Movies":
+            response = requests.get(MovieRequest.TOP_RATED)
+            info_to_display = ["title", "release_date", "Movies"]
+        elif menu_item == "TV Shows":
+            response = requests.get(MovieRequest.TOP_RATED_TV)
+            info_to_display = ["name", "first_air_date", "TV Shows"]
+    else:
+        # Search
+        if menu_item == "Movies":
+            response = requests.get(MovieRequest.search_movie(query))
+            info_to_display = ["title", "release_date", "Movies"]
+        elif menu_item == "TV Shows":
+            response = requests.get(MovieRequest.search_tv(query))
+            info_to_display = ["name", "first_air_date", "TV Shows"]
+
+    movies_data = response.json()
+    movies_data = movies_data["results"]
+
+    return render_template(
+        "flixlist.html",
+        title="FlixList",
+        menu_item=menu_item,
+        movies_data=movies_data,
+        info_to_display=info_to_display,
+    )
+
 
 @app.route("/friendlist")
 def friendflix():
```

Comment:
I changed the function name to **friendlist**. A conflict may arise.

Reply:
Got it, we should keep your function name changes when merging